Content.Server/BugReports/BugReportManager.cs

```diff
@@ -0,0 +1,158 @@
+using Robust.Shared;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+using Robust.Shared.Timing;
+
+namespace Content.Server.BugReports;
+
+/// <summary>
+///     Simple manager to handle players bug reports. Will forward valid reports through <see cref="ValidPlayerBugReportReceived"/>
+/// </summary>
+public sealed class BugReportManager : IBugReportManager, IPostInjectInit
+{
+    [Dependency] private readonly IServerNetManager _net = default!;
+    [Dependency] private readonly IEntityManager _entity = default!;
+    [Dependency] private readonly PlayTimeTrackingManager _playTime = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly IAdminLogManager _admin = default!;
+    [Dependency] private readonly IGameMapManager _map = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly ILogManager _log = default!;
+
+    private ISawmill _sawmill = default!;
+
+    // Listen to this event handler if you want to get alerted anytime a players valid bug report comes in!
+    public event EventHandler<ValidPlayerBugReportReceivedEvent>? ValidPlayerBugReportReceived;
+
+    /// <summary>
+    ///     List of player NetIds and the number of bug reports they have submitted this round.
+    ///     UserId -> (bug reports this round, last submitted bug report)
+    /// </summary>
+    private readonly Dictionary<NetUserId, (int, DateTime)> _bugReportsPerPlayerThisRound = new();
+
+    public void Initialize()
+    {
+        _net.RegisterNetMessage<BugReportMessage>(ReceivedPlayerBugReport);
+    }
+
+    private void ReceivedPlayerBugReport(BugReportMessage message)
+    {
+        var netId = message.MsgChannel.UserId;
+
+        if (!CheckIfValid(message, netId))
+            return;
+
+        _bugReportsPerPlayerThisRound[netId] = (_bugReportsPerPlayerThisRound.GetValueOrDefault(netId).Item1 + 1, DateTime.UtcNow);
+
+        var title = message.ReportInformation.BugReportTitle;
+        var description = message.ReportInformation.BugReportDescription;
+
+        _admin.Add(LogType.BugReport, LogImpact.High, $"{message.MsgChannel.UserName}, {netId}: submitted a bug report. Title: {title}, Description: {description}");
+
+        var bugReport = CreateBugReport(message);
+
+        ValidPlayerBugReportReceived?.Invoke(this, bugReport);
+    }
+
+    // When the round restarts, clear the dictionary.
+    public void Restart()
+    {
+        _bugReportsPerPlayerThisRound.Clear();
+    }
+
+    /// <summary>
+    ///     Checks that the given report is valid. Includes various checks on the report and the player.
+    /// </summary>
+    /// <returns>True if the report is valid, false there is an issue with the report or the user sending it.</returns>
+    private bool CheckIfValid(BugReportMessage message, NetUserId netId)
```

Comment:
Nit but some of this logic is checking if the report itself is invalid (e.g. title too long) and some of it is checking if a report can currently be sent by the player.